### ISSUE TITLE
Working with StickyListHeaders

### ISSUE THREAD

**Issue by aherrick** [wdullaer/SwipeActionAdapter#29]
Hi,

I'm trying to use swipe adapter with StickyListHeaders.
https://github.com/emilsjolander/StickyListHeaders

I think I'm close to a solution, however when I swipe on a row I get the following immediate exception. It does however hit the "hasActions" method

```
08-02 11:33:07.364    1655-1655/com.slaptap.tappedin E/InputEventReceiver﹕ Exception dispatching input event.
08-02 11:33:07.364    1655-1655/com.slaptap.tappedin E/MessageQueue-JNI﹕ Exception in MessageQueue callback: handleReceiveCallback
08-02 11:33:07.380    1655-1655/com.slaptap.tappedin E/MessageQueue-JNI﹕ java.lang.NullPointerException
            at com.wdullaer.swipeactionadapter.SwipeActionTouchListener.onTouch(SwipeActionTouchListener.java:419)
            at android.view.View.dispatchTouchEvent(View.java:7701)
            at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2210)
            at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1945)
```

I am setting things up like so inside of a Fragment 

```
            mAdapter = new VenueBeerListAdapter(getActivity(), data);

            listView.setAdapter(mAdapter);
            
            swipeAdapter = new SwipeActionAdapter(mAdapter);

            swipeAdapter.setListView(listView.getWrappedList())
                    .setSwipeActionListener(vbf)
                    .addBackground(SwipeDirections.DIRECTION_FAR_LEFT, R.layout.view_venue_beer_mark_blown)
                    .addBackground(SwipeDirections.DIRECTION_NORMAL_LEFT, R.layout.view_venue_beer_mark_blown)
                    .addBackground(SwipeDirections.DIRECTION_FAR_RIGHT, R.layout.view_venue_beer_mark_seen)
                    .addBackground(SwipeDirections.DIRECTION_NORMAL_RIGHT, R.layout.view_venue_beer_mark_seen).notifyDataSetChanged();
```

Any initial thoughts on what I might be doing wrong? I can provide more info if need.. really appreciate the help :)

**Comment by wdullaer** [comment]
You need to set the SwipeActionAdapter to your ListView rather than your own Adapter.

Second line of your second code sample is probably the cause of the error.

**Comment by aherrick** [comment]
Hi,

Thanks for the response. The problem I have with that is StickListHeaders has its own Adapter type. 
How do you think I should handle that?

![8-2-2015 7-47-45 pm](https://cloud.githubusercontent.com/assets/586671/9028696/718d66b2-394f-11e5-94da-075b4cce8f69.png)

**Comment by wdullaer** [comment]
Take a look at #28 

These type of issues are also better handled on stackexchange as they are not really an issue with this library.
I should create a tag for this library so I can more easily keep track of questions there.

**Comment by aherrick** [comment]
@wdullaer 

Thanks, however I've implemented what you suggested for #28 and I'm still receiving the same error as I had originally.

I have my Base Adapter wrapped by the Swipe Adapter. I then Have another adapter which extends Decorator Adapter and implements Sticky Adapater.

```
       mAdapter = new ListAdapter(getActivity(), data);
            swipeAdapter = new SwipeActionAdapter(mAdapter);
            ListStickyAdapter vbsa = new ListStickyAdapter(swipeAdapter);
            listView.setAdapter(vbsa);

            swipeAdapter.setListView(listView.getWrappedList())
```

Does it have anything to do with the fact that I'm having to use the WrappedList to pass to the swipe adapter? Should I be callilng "setListView" on the swipeAdapater?

I feel I'm very close on this... thank you for your help
